### PR TITLE
Feature/ task type, task status & asset type descriptions

### PR DESCRIPTION
**Pull request by scottmcdonnell**
**Problem**
With some complex project pipelines we had amassed a large collection of Task Types, Statuses and Asset Types. 
It is not always clear the distinctions between each, and the intention of the items on creation. 
An additional Description field would allow the items to be documented clearly.

**Solution**
Adding descriptions to the Lists for Task Types, Task Statuses and Asset Types. 

Task Status Edit:
image

Task Status List:
Description is added a ? icon, with a rollover tooltip, beside the name of the status if set. 
image

Descriptions are exported in the CSV.

Asset Type List:
Descriptions as rollover if set. Exported to CSV.
image

Task Type List:
Descriptions as table cell as the name field is already a component. Exported to CSV.
image

this PR requires extra description fields in Zou - see https://github.com/cgwire/zou/pull/824

### CODE REVIEW

**Comment by EvanBldy** [comment]
Hi @scottmcdonnell, 
Thanks for your PRs, we will merge them this week (we need to do a release before).

File: src/components/modals/EditTaskStatusModal.vue [hunk 1]
```diff
@@ -36,6 +36,14 @@
             v-focus
```
Comment (NicoPennec) [inline src/components/modals/EditTaskStatusModal.vue:36]: useless directive.
v-focus is already set on the name field.
```suggestion
```

File: src/components/modals/EditTaskTypeModal.vue [hunk 2]
```diff
@@ -35,6 +35,13 @@
             @enter="confirmClicked"
             v-model="form.allow_timelog"
           />
+          <textarea-field
+            ref="descriptionField"
+            :label="$t('task_types.fields.description')"
+            v-model="form.description"
+            @enter="confirmClicked"
+            v-focus
```
Comment (NicoPennec) [inline src/components/modals/EditTaskTypeModal.vue:43]: useless directive.
v-focus is already set on the name field.
```suggestion
```

File: src/components/modals/EditAssetTypeModal.vue [hunk 3]
```diff
@@ -31,6 +31,12 @@
             :maxlength="30"
             v-model="form.short_name"
             @enter="runConfirmation"
+          >
+          <textarea-field
+            ref="descriptionField"
```
Comment (NicoPennec) [inline src/components/modals/EditAssetTypeModal.vue:36]: `ref` never used
```suggestion
```

File: src/components/modals/EditTaskStatusModal.vue [hunk 4]
```diff
@@ -36,6 +36,14 @@
...
             v-if="taskStatusToEdit.short_name !== 'todo'"
           />
+          <textarea-field
+            ref="descriptionField"
```
Comment (NicoPennec) [inline src/components/modals/EditTaskStatusModal.vue:40]: `ref` never used
```suggestion
```

File: src/components/modals/EditTaskTypeModal.vue [hunk 5]
```diff
@@ -35,6 +35,13 @@
             @enter="confirmClicked"
             v-model="form.allow_timelog"
           />
+          <textarea-field
+            ref="descriptionField"
```
Comment (NicoPennec) [inline src/components/modals/EditTaskTypeModal.vue:39]: `ref` never used
```suggestion
```

File: src/components/modals/EditAssetTypeModal.vue [hunk 6]
```diff
@@ -31,6 +31,12 @@
             :maxlength="30"
             v-model="form.short_name"
             @enter="runConfirmation"
+          >
```
Comment (NicoPennec) [inline src/components/modals/EditAssetTypeModal.vue:34]:
```suggestion
          />
```

**Comment by NicoPennec** [comment]
@scottmcdonnell
Thanks for your contribution 🙏 
It will be available in the next release!

**Pull request merged**